### PR TITLE
feat: Adding new mnemonic wallet provider for e2e testing

### PR DESCRIPTION
**Pull request by aorumbayev**
### Proposed changes:

- New provider called mnemonic, similar to kmd it prompts user for 25 word passphrase that upon entering is stored as an account record with addr and pk - passphrase itself isn't persisted.
- Adding prettierrc file to make sure folks contributing and using prettier won't have conflicts with indent style used in codebase

### How to test?

Refer to the forked main branch - execute `yarn install` and `yarn dev` to launch the playbook. Make sure to explicitly add it to playbook because similarly to kmd its excluded from default list of providers. 

### Who may benefit from this provider?

Any folks who has integration with use-wallet and would like to have a simple way to perform e2e test, with mnemonic provider like this user can automate pasting the phrase to the input prompt in most automation libraries out there. In contrast e2e test with real walletconnect/myalgowallet or etc gets tricker as the amount of complexity to automate it is far greater.

### Further notes

There were no unit tests setup in the repo so instead i tested it locally so far - will use this forked version in my swapper.algoworld.io project for the time being to have a more specific justification on why this can be helpful. Feel free to leave any comments/suggestions - would love to address those

### CODE REVIEW

File: src/clients/mnemonic/client.ts
Comment (gabrielkuettel) [inline src/clients/mnemonic/client.ts:97]:
```suggestion
    connectedAccounts: string[],
    transactions: Uint8Array[],
    indexesToSign?: number[],
    returnGroup = true
```

Add the `indexesToSign` and `returnGroup` param. 

The `indexesToSign` param lets you specify an array of indexes in the atomic group that should be signed. It's modeled after the `Signer` interface specified here: https://github.com/algorand/js-algorand-sdk/blob/develop/src/signer.ts#L15-L18

The `returnGroup` option sets whether or not the entire transaction group is returned, or only the transactions that were signed. Defaults to `true` to return the entire group.

File: src/clients/mnemonic/client.ts
Comment (gabrielkuettel) [inline src/clients/mnemonic/client.ts:131]:
```suggestion
   for (const idx in decodedTxns) {
      const dtxn = decodedTxns[idx];
      const isSigned = "txn" in dtxn;

      // push the incoming txn into signed, we'll overwrite it later
      signedTxns.push(transactions[idx]);

      // Its already signed, skip it
      if (isSigned) {
        continue;
        // Not specified in indexes to sign, skip it
      } else if (
        indexesToSign &&
        indexesToSign.length &&
        !indexesToSign.includes(Number(idx))
      ) {
        continue;
      }
      // Not to be signed by our signer, skip it
      else if (
        !connectedAccounts.includes(this.algosdk.encodeAddress(dtxn.snd))
      ) {
        continue;
      }

      // overwrite with an empty blob
      signedTxns[idx] = new Uint8Array();

      const txn = this.algosdk.Transaction.from_obj_for_encoding(dtxn);
      const signedTxn = txn.signTxn(this.#client.sk);
      signingResults.push(signedTxn);
    }
```

Implement the `indexesToSign` param.

File: src/clients/mnemonic/client.ts
Comment (gabrielkuettel) [inline src/clients/mnemonic/client.ts:144]:
```suggestion
        const formattedTxns = signedTxns.reduce<Uint8Array[]>((acc, txn, i) => {
      // If its an empty array, infer that it is one of the
      // ones we wanted to have signed and overwrite the empty buff
      if (txn.length === 0) {
        acc.push(signingResults[signedIdx]);

        signedIdx += 1;
      } else if (returnGroup) {
        acc.push(txn);
      }

      return acc;
    }, []);

    return Promise.resolve(formattedTxns);
```

Implement the `returnGroup` param.

**Comment by gabrielkuettel** [comment]
Thank you for this outstanding contribution! We agree this will be enormously helpful for supporting E2E tests and are looking forward to get this merged and writing some tests around this. 

I think this looks good but since you made your PR, there have been some updates to the `signTransactions` method that need to be incorporated here before it can be merged (my fault). I've added them as comments/code suggestions. If you could take a look at this, I think we're ready. @drichar may also want to give final approval, as he wanted to try out some E2E tests, but we'll prioritize this to get it merged fast. 

Let me know if you have any Qs on my comments.

**Review by gabrielkuettel — APPROVED** [review]

**Review by gabrielkuettel — CHANGES_REQUESTED** [review]
Requesting changes. See: https://github.com/TxnLab/use-wallet/pull/28#issuecomment-1384785488

**Comment by aorumbayev** [comment]
@gabrielkuettel please expect me to address the PR comments before EOD tomorrow. Ill keep you updated and thanks for the review!

**Comment by aorumbayev** [comment]
@gabrielkuettel @drichar all good on my side re-tested component in storybook, changes applied are working ok

File: .prettierrc
Comment (gabrielkuettel) [inline .prettierrc:6]:
```suggestion
{
  "semi": false,
  "singleQuote": true,
  "printWidth": 100,
  "trailingComma": "none"
}
```

Comment (gabrielkuettel) [reply .prettierrc:6]: Appreciate you adding this, it was needed. The suggested changes aligns this with our internal repos.

**Review by gabrielkuettel — CHANGES_REQUESTED** [review]
Tested this and everything is working great. Fantastic work. If you could please commit my suggested change regarding the prettier config, then this is good to go and we will merge.

**Review by gabrielkuettel — APPROVED** [review]

**Pull request merged**
